src/components/VersionRestartModal/VersionRestartModal.tsx

```diff
@@ -139,43 +136,115 @@ const VersionRestartModal: React.FC<Props> = ({
       ]}
       data-cy="version-restart-modal"
     >
+      <VersionTasks
+        version={version}
+        selectedTasks={selectedTasks}
+        setBaseStatusFilterTerm={setVersionBaseStatus(version?.id)}
+        setPatchStatusFilterTerm={setVersionStatus(version?.id)}
+        toggleSelectedTask={toggleSelectedTask}
+        baseStatusFilterTerm={baseStatusFilterTerm[version?.id]}
+        patchStatusFilterTerm={patchStatusFilterTerm[version?.id]}
+      />
+
+      {childVersions && (
+        <div data-cy="select-downstream">
+          <ConfirmationMessage weight="medium" data-cy="confirmation-message">
+            Downstream Tasks
+          </ConfirmationMessage>
+          {childVersions?.map((v) => (
+            <Accordion
+              key={v?.id}
+              title={
+                <BoldTextStyle>
+                  {v?.projectIdentifier ? v?.projectIdentifier : v?.project}
+                </BoldTextStyle>
+              }
+              contents={
+                <TitleContainer>
+                  <VersionTasks
+                    version={v}
+                    selectedTasks={selectedTasks}
+                    setBaseStatusFilterTerm={setVersionBaseStatus(v?.id)}
+                    setPatchStatusFilterTerm={setVersionStatus(v?.id)}
+                    toggleSelectedTask={toggleSelectedTask}
+                    baseStatusFilterTerm={baseStatusFilterTerm[v.id]}
+                    patchStatusFilterTerm={patchStatusFilterTerm[v.id]}
+                  />
+                </TitleContainer>
+              }
+            />
+          ))}
+          <br />
+        </div>
+      )}
+
+      <ConfirmationMessage weight="medium" data-cy="confirmation-message">
+        Are you sure you want to restart the {selectedTotal} selected tasks?
```

Comment:
[nit] Would be nice for this to read "task" if there's only one task selected.

Reply:
yes, that would be nice. But since it's already been this way for a while and I'm in a bit of a rush with it, maybe this can be a future improvement?

Reply:
This is a pretty common pattern we use in allot of places. I opened a ticket so we could add a util method for this in the future. https://jira.mongodb.org/browse/EVG-15437